updater/spec/dependabot/updater/operations/update_all_versions_spec.rb

```diff
@@ -0,0 +1,127 @@
+      repo: "dependabot-fixtures/dependabot-test-ruby-package",
+      directory: "/",
+      branch: nil,
+      api_endpoint: "https://api.github.com/",
+      hostname: "github.com"
+    )
+  end
+
+  let(:job) do
+    instance_double(Dependabot::Job,
+                    package_manager: "emptyreturn",
+                    repo_contents_path: nil,
+                    credentials: [],
+                    reject_external_code?: false,
+                    source: source,
+                    dependency_groups: [],
+                    allowed_update?: true,
+                    experiments: { large_hadron_collider: true },
+                    ignore_conditions_for: [">= 0"],
+                    security_advisories_for: [],
+                    log_ignore_conditions_for: [],
+                    existing_pull_requests: [],
+                    requirements_update_strategy: nil)
+  end
+
+  let(:dependency_snapshot) do
+    Dependabot::DependencySnapshot.create_from_job_definition(
+      job: job,
+      job_definition: job_definition_with_fetched_files
+    )
+  end
+
+  let(:job_definition_with_fetched_files) do
+    job_definition.merge({
+      "base_commit_sha" => "mock-sha",
+      "base64_dependency_files" => encode_dependency_files(dependency_files)
+    })
+  end
+
+  let(:dependency_files) do
+    original_bundler_files
+  end
+
+  let(:mock_error_handler) do
+    instance_double(Dependabot::Updater::ErrorHandler)
+  end
+
+  before do
+    allow(Dependabot.logger).to receive(:info)
+    stub_request(:get, "https://rubygems.org/api/v1/gems/business.json")
+      .to_return(status: 200, body: fixture("ruby/rubygems_response.json"))
+    stub_request(:get, "https://rubygems.org/api/v1/versions/business.json")
+      .to_return(status: 200, body: fixture("ruby/versions-business.json"))
+  end
+
+  after do
+    Dependabot::Experiments.reset!
+  end
+
+  context "when updatedDeps is empty" do
+    let(:job_definition) do
+      {
+        "type" => "update",
+        "package_manager" => "bundler",
+        "target_dependency" => "business",
+        "updatedDeps" => []
+      }
+    end
+
+    let(:dependency) do
+      Dependabot::Dependency.new(
+        name: dependency_name,
+        version: dependency_version,
+        previous_version: dependency_previous_version,
+        requirements: requirements,
+        previous_requirements: previous_requirements,
+        package_manager: "emptyreturn"
+      )
+    end
+
+    let(:dependency_name) { "business" }
+    let(:dependency_version) { "1.5.0" }
+    let(:dependency_previous_version) { "1.5.0" }
+    let(:requirements) do
+      [{ file: "Gemfile", requirement: "~> 1.5.0", groups: [], source: nil }]
+    end
+    let(:previous_requirements) do
+      [{ file: "Gemfile", requirement: "~> 1.5.0", groups: [], source: nil }]
+    end
+
+    it "does not update any dependencies" do
+      expect { subject.send(:check_and_create_pull_request, dependency) }.to raise_error(
```

Comment:
We don't usually test private methods directly. That said it looks like if you take this suggestion you'll need to change this test a bit to get it to work. DependencySnapshot wants something it can parse, might be nice if there was a way to create one and inject a fake dependency.

```suggestion
      expect { subject.perform }.to raise_error(
```

Reply:
Yeah, I was originally trying to call `perform` (if only because I didn't know you could do `.send` for private methods in Ruby) and it ended up needing a lot of extra context/mocks that wasn't directly related to the change we had made.

In the absence of existing tests of this class or a similar example for parsing-mocks I figured it was better to do the simpler version of things, and when someone adds more tests to this area maybe it makes sense to switch it over. I can add a comment to that effect if it helps. Does that allay your concerns?